### ISSUE TITLE
Autofill for TextField is not firing onChange in all browsers

### ISSUE THREAD

**Issue by bensontrent** [stripe/stripe-apps#980]
This may not be a Stripe Apps issue, but it's an issue we've had to deal with in our application.  When a user inputs their origin address for shipping, depending on their browser, the onChange event does not get fired when auto-filling their address via their browser.  And in some browsers, the `autofill` attribute is not respected.  We would like them to be able to auto-fill their address, and we would like the onChange event to get respected, regardless of the browser.  For reference see this issue: https://stackoverflow.com/questions/55244590/autofill-does-not-trigger-onchange

I've created a test component that should work similarly across browsers, but it does not behave as expected in cross-browser experiments:

![image](https://github.com/user-attachments/assets/07ba631f-4a93-4215-904a-e577611a5081)

```tsx

import {
  Box,
  Button,
  ContextView,
  Switch,
  TextField,
} from "@stripe/ui-extension-sdk/ui";
import type { ExtensionContextValue } from "@stripe/ui-extension-sdk/context";
import { useCallback, useEffect, useRef, useState } from "react";

type Address = {
  name?: string;
  city?: string;
  state?: string;
  line1?: string;
}

const App = () => {
  const [address, setAddress] = useState<Address>({})
  const [autofill, setAutofill] = useState<boolean>(false)
  const [displayAddress, setDisplayAddress] = useState<Address>()

  const currentAddress = useRef<Address>({})

  const showAddressResult = useCallback(() => {
    setDisplayAddress(currentAddress.current)
  }, [])

  useEffect(() => {
    currentAddress.current = address
  }, [address])

  const handleChange = (key: keyof Address, value: string) => {
    setAddress(prev => ({ ...prev, [key]: value }));

  };

  return (
    <ContextView title="Autofill address browser test">
      <Box css={{  stack: "y", gap: "small" }}>
        This applications tests if auto-filled forms trigger an on-change event, also reveals if the autofill attribute is respected by the browser.
      

      <Switch label="Set auto fill attribute to on" defaultChecked={autofill} onChange={(e) => setAutofill(e.target.checked)} />
      <TextField
        label="Name" autoComplete={autofill ? "on" : "off"} placeholder="Name" name="name"
        onChange={(e) => handleChange("name", e.target.value)}
      />

<TextField
        label="Address Line 1" autoComplete={autofill ? "on" : "off"} placeholder="Address line 1" name="line1"
        onChange={(e) => handleChange("line1", e.target.value)}
      />

      <TextField
        label="City" autoComplete={autofill ? "on" : "off"} placeholder="City" name="city"
        onChange={(e) => handleChange("city", e.target.value)}
      />
            <TextField
        label="State" autoComplete={autofill ? "on" : "off"} placeholder="State" name="state"
        onChange={(e) => handleChange("state", e.target.value)}
      />


      <Button type="primary" onPress={() => showAddressResult()}>
        Show result
      </Button>


      {displayAddress && <Box>
        <Box>
          Name: {displayAddress.name}
        </Box>
        <Box>
          Address Line 1: {displayAddress.line1}
        </Box>
        <Box>
          City: {displayAddress.city}
        </Box>
        <Box>
          State: {displayAddress.state}
        </Box>
      </Box>}

      </Box>

    </ContextView>
  );
};

export default App;


```

**Comment by ngrubb-stripe** [comment]
Thanks for the report, @bensontrent!

We have tested this functionality according to [https://docs.stripe.com/dashboard/basics#browser-compatibility](https://docs.stripe.com/dashboard/basics#browser-compatibility). We couldn't reproduce this on any of our supported browsers, the `onChange` was triggered in all supported browsers on autocomplete. Seeing as this issue isn't browser behavior, the Apps platform does not plan to solve this issue.

**Comment by bensontrent** [comment]
Hi @ngrubb-stripe I know it's been a while, but I wanted to tell you that I was wrong, and you were right.  When I embarked on my Stripe Apps React project, I was woefully unfamiliar with the intricacies of React, and now that I am painfully aware of my shortcomings, I wish to apologize to you for sending you on this wild goose chase.  I found my error (with more experience) in 2 lines of code that were so blatantly obvious that I'm embarrassed to have the original lines of 2 lines of code on my private Github.  It had to do with using (...value, ...value) spread operators for new values when using blank values for the latest updated properties.  Cheers, and thank you for your excellent work!
